core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -271,6 +275,49 @@ object TopicCommand extends Logging {
     ret.toMap
   }
 
+  private[admin] class PartitionsDescription(topic: String, partitions: List[(Int, Seq[Int])], liveBrokers: Set[Int], topicAndPartitionToLeaderAndIsr: Map[(String, Int), (Option[Int], Seq[Int])]) {
+
+    def describeAndPrint(): String = {
+      val sb = new StringBuilder
+      for ((partitionId, assignedReplicas) <- partitions) {
+        val (leader, inSyncReplicas) = topicAndPartitionToLeaderAndIsr.get(topic, partitionId).get
+        if (shouldPrint(assignedReplicas, leader, inSyncReplicas)) {
+          sb.append(printIt(topic, partitionId, assignedReplicas, leader, inSyncReplicas))
+        }
+      }
+      sb.toString()
+    }
+
+    def shouldPrint(assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): Boolean = true
+
+    def printIt(topic: String, partitionId: Int, assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): String = {
+      val sb = new StringBuilder
+      sb.append("\tTopic: " + topic)
+      sb.append("\tPartition: " + partitionId)
+      sb.append("\tLeader: " + (if (leader.isDefined) leader.get else "none"))
+      sb.append("\tReplicas: " + assignedReplicas.mkString(","))
+      sb.append("\tIsr: " + inSyncReplicas.mkString(","))
+
+      val description = sb.toString()
+      println(description)
+      description
+    }
+  }
+
+  private[admin] class UnderReplicatedPartitionsDescription(topic: String, partitions: List[(Int, Seq[Int])], liveBrokers: Set[Int], topicAndPartitionToLeaderAndIsr: Map[(String, Int), (Option[Int], Seq[Int])])
```

Comment:
I think the class hierarchy is a bit of an overkill... 
having a shouldPrint method with two-level boolean logic is not that bad.